### PR TITLE
feat: add version tag to process instance header

### LINKED ISSUE #21170: Add version tag to process instance details header
### User Story

```Gherkin
Scenario: As an Operate user
    Given I am process instance details view 
    Then I can see the related version tag in the instance header
```
### Designs
https://www.figma.com/design/7lJFTkWKfhQUfM6nfil4Bh/Version-tag?node-id=41-28180&t=VrnT07nWiWiK23nD-1

### Acceptance Criteria

- [x] If there is a version tag associated to a process, it will be shown in the instance header
- [x] If there is no version tag, the version tag column will not be visible in the header
- [x] The version tag has a tooltip with a description (see designs for text)

### Definition of Ready - Checklist

- [x] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered and added to the product-hub issue risk assessment
- [ ] Cross-team dependencies have been considered

Optional:
- [ ] Design input has been collected by the assignee
### :point_right: Handover Dev to QA 

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Versions to validate:
> Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released)

### PR DESCRIPTION
**Pull request by pedesen**
## Description

- add support for processes API endpoint 
- add feature flag
- add API mock
- add Version tag to process instance header
- add tests, fix existing ones

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #21170

### CODE REVIEW

File: operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.setup.ts [hunk 1]
```diff
@@ -1,35 +0,0 @@
-/*
```
Comment (pedesen) [inline operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.setup.ts:1]: This file was renamed to `index.setup.tsx`

File: operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.setup.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+const mockProcess = {
+  id: '2251799813688076',
+  name: 'Complex Process',
+  version: 3,
+  bpmnProcessId: 'complexProcess',
+  versionTag: 'myVersionTag',
+};
+
+const Wrapper: React.FC<{children?: React.ReactNode}> = ({children}) => {
```
Comment (pedesen) [inline operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.setup.tsx:46]: This wrapper was added here and is now reused by index.test and versionTag.test

File: operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.tsx [hunk 3]
```diff
@@ -23,59 +25,80 @@ import {InstanceHeader} from 'modules/components/InstanceHeader';
...
+  },
+  {
+    name: 'Parent Process Instance Key',
+    skeletonWidth: '142px',
+  },
+  {
+    name: 'Called Process Instances',
+    skeletonWidth: '142px',
+  },
+] as const;
```
Comment (pedesen) [inline operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.tsx:76]: I tried to make this a bit simpler. Please let me know what you think or if you have some ideas to further improve it.

Comment (vitorwtavares) [reply operate/client/src/App/ProcessInstance/ProcessInstanceHeader/index.tsx:76]: I think it is a bit simpler with the changes, not sure if we can do much more. The structure itself is a bit confusing, but this looks good to me.

**Review by vitorwtavares — APPROVED** [review]

**Review by gastonpillet01 — APPROVED** [review]
Version tag was correclty added to process instance header

**Pull request merged**